src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+    if (this.health <= 0) {
+      this.health = 0;
+      this.remove();
+    }
+  }
+
+  remove() {
+    const index = Animal.alive.indexOf(this);
+    Animal.alive.splice(index, 1);
```

Comment:
The current implementation of the `remove` method uses `indexOf` and `splice` to remove the animal from the `Animal.alive` array, which can be inefficient due to the nesting of loops. Consider using the `filter` method to create a new array that excludes the deceased animal, which avoids modifying the array during iteration and can be more performant.